app/lib/features/profile/pages/my_profile_page.dart

```diff
@@ -129,56 +50,66 @@ class MyProfilePage extends ConsumerWidget {
       type: FileType.image,
     );
     if (result != null) {
+      EasyLoading.show(status: 'Updating profile image');
+
       final file = result.files.first;
       await profile.account.uploadAvatar(file.path!);
+      ref.invalidate(accountProfileProvider);
 
       if (!context.mounted) {
         return;
       }
-      customMsgSnackbar(context, 'Avatar uploaded');
+      // close loading
+      EasyLoading.dismiss();
     } else {
       // user cancelled the picker
     }
   }
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    final account = ref.watch(accountProfileProvider);
-    final size = MediaQuery.of(context).size;
+  Widget build(BuildContext context) {
+    return WithSidebar(
+      sidebar: const SettingsPage(),
+      child: Scaffold(
+        appBar: _buildAppbar(context),
+        body: _buildBody(context),
+      ),
+    );
+  }
 
-    return account.when(
-      data: (data) {
-        final userId = data.account.userId().toString();
-        return Scaffold(
-          appBar: AppBar(
-            title: const Text('My profile'),
-          ),
-          body: SingleChildScrollView(
-            child: Column(
-              children: [
-                Container(
-                  margin: const EdgeInsets.fromLTRB(20, 15, 20, 0),
-                  decoration: BoxDecoration(
-                    borderRadius: BorderRadius.circular(25),
-                  ),
-                  width: double.infinity,
-                  child: ClipRRect(
-                    borderRadius: BorderRadius.circular(25),
-                    child: const SizedBox(),
-                  ),
-                ),
-                Container(
-                  constraints: const BoxConstraints(maxWidth: 400),
-                  margin: const EdgeInsets.symmetric(horizontal: 20),
-                  alignment: Alignment.center,
-                  child: Column(
-                    mainAxisAlignment: MainAxisAlignment.center,
-                    crossAxisAlignment: CrossAxisAlignment.center,
-                    children: [
-                      GestureDetector(
-                        onTap: () => updateAvatar(data, context, ref),
+  AppBar _buildAppbar(BuildContext context) {
+    return AppBar(
+      title: Text(
+        'Profile',
+        style: Theme.of(context).textTheme.titleLarge,
+      ),
+    );
+  }
+
+  Widget _buildBody(BuildContext context) {
+    return Consumer(
```

Comment:
We don't need this ```Consumer``` wrapper do we? Could be simply like :
```suggestion
    Widget _buildBody(BuildContext context){
    final account = ref.watch(accountProfileProvider);
    return account.when(
    ...
    }
```